torch/nn/modules/module.py

```diff
@@ -1407,6 +1410,27 @@ def _register_load_state_dict_pre_hook(self, hook, with_module=False):
         self._load_state_dict_pre_hooks[handle.id] = hook
         return handle
 
+    def _register_load_state_dict_post_hook(self, hook, with_module=False):
```

Comment:
curious why "with_module" option is needed?

Reply:
Looks like this was done to match `_register_load_state_dict_pre_hook()`, where the `with_module` flag was added later on.

Reply:
Note that we wrote a full issue on how we want these hooks to happen with Rohan. This PR is outdated wrt that and should be updated to match.